packages/dynamite/dynamite/lib/src/builder/generate_ofs.dart

```diff
@@ -0,0 +1,243 @@
+import 'package:code_builder/code_builder.dart';
+import 'package:dynamite/src/builder/state.dart';
+import 'package:dynamite/src/helpers/dart_helpers.dart';
+import 'package:dynamite/src/models/openapi.dart' as openapi;
+import 'package:dynamite/src/models/type_result.dart';
+
+Iterable<Spec> buildOfsExtensions(
+  final openapi.OpenAPI spec,
+  final State state,
+) sync* {
+  final typeResults = state.resolvedTypes.whereType<TypeResultSomeOf>().toSet();
+
+  for (final result in typeResults) {
+    yield TypeDef((final b) {
+      b
+        ..name = result.name
+        ..definition = refer(result.dartType.name);
+    });
+
+    if (result.optimizedSubTypes.length <= 1) {
+      continue;
+    }
+    yield* generateSomeOf(result);
+  }
+}
+
+Iterable<Spec> generateSomeOf(
+  final TypeResultSomeOf result,
+) sync* {
+  final identifier = result.name;
+  final results = result.optimizedSubTypes;
+  final serializerName = '_\$${identifier}Serializer';
+
+  final fields = <TypeResult, String>{};
+  for (final result in results) {
+    final dartType = result.nullableName;
+    final dartName = toDartName(dartType);
+    fields[result] = toFieldName(dartName, result.className);
+  }
+
+  final values = Method((final b) {
+    b
+      ..returns = refer('List<dynamic>')
+      ..type = MethodType.getter
+      ..name = '_values'
+      ..lambda = true
+      ..body = Code('[${fields.values.join(',')}]');
+  });
+
+  final oneOfValidator = Method((final b) {
+    b
+      ..name = 'validateOneOf'
+      ..returns = refer('void')
+      ..lambda = true
+      ..body = const Code('dynamite_utils.validateOneOf(_values)');
+  });
+
+  final anyOfValidator = Method((final b) {
+    b
+      ..name = 'validateAnyOf'
+      ..returns = refer('void')
+      ..lambda = true
+      ..body = const Code('dynamite_utils.validateAnyOf(_values)');
+  });
```

Comment:
Instead of adding these methods to the interfaces the relevant places could just call the methods directly.

Edit 1: I see that the _values field is private, is that why you added these proxy methods to not expose the field directly?

Edit 2: Since the values in the _values field are not private it would be possible to drop the additonal methods and at the invocation dynamically create the list of arguments. Honestly I don't care which of these options you chose in the end :woman_shrugging:

Reply:
I quite like calling validate directly on the type.
Imagine:
```dart
var myRecord = (10, 0.5, 'test');
myRecord.validateOneOf();
``` 

I'll add this to the validation hook of the object having the record.
This is needed as the records are currently not validated at deserialization as a serializer can be used for both `oneOf`s and `anyOf`s.

Reply:
Anything that needs to be changed or is this done?

Reply:
The validation is still missing and just adding it will lead to multiple extension methods being declared on the same type.
https://dart.dev/tools/diagnostic-messages#ambiguous_extension_member_access

Should be trivial to fix but probably something I'll do on monday.